stdlib/public/core/Integers.swift.gyb

```diff
@@ -2204,6 +2206,56 @@ ${assignmentOperatorComment(x.operator, False)}
 }
 
 //===----------------------------------------------------------------------===//
+//===--- DefaultWords -----------------------------------------------------===//
+//===----------------------------------------------------------------------===//
+
+/// A collection containing the words of the binary representation
+/// of an arbitrary fixed-width integer type, in order from the least
+/// significant to most significant.
+///
+/// The words of negative values are in two's complement representation,
+/// regardless of the integer type's underlying implementation.
+public struct DefaultWords<Integer: FixedWidthInteger> : RandomAccessCollection {
+  public typealias Indices = CountableRange<Int>
+  public typealias SubSequence = RandomAccessSlice<DefaultWords<Integer>>
+
+  var _value: Integer
+
+  public init(_ value: Integer) {
+    self._value = value
+  }
+
+  public var count: Int {
+    return (Integer.bitWidth + ${word_bits} - 1) / ${word_bits}
+  }
+
+  public var startIndex: Int { return 0 }
+
+  public var endIndex: Int { return count }
+
+  public subscript(position: Int) -> UInt {
+    get {
+      _precondition(position >= 0, "Negative word index")
+      _precondition(position < endIndex, "Word index out of range")
+      let shift = UInt(position._value) &* ${word_bits}
+      _sanityCheck(shift < UInt(_value.bitWidth._value))
+      return (_value &>> Integer(_truncatingBits: shift))._lowWord
+    }
+  }
+}
+
+extension FixedWidthInteger where Words == DefaultWords<Self> {
```

Comment:
I think that having the `typealias Words` and this conditional expression is unnecessarily complicated to track. Why not have them grouped together in the same non-conditional extension?